.github/actions/build-module/action.yml

```diff
@@ -0,0 +1,44 @@
+name: 'Build Module'
+description: 'Builds the viamrtsp module using Docker'
+inputs:
+  target_os:
+    description: 'Target OS'
+    required: true
+  target_arch:
+    description: 'Target architecture'
+    required: true
+  docker_image:
+    description: 'Docker image to use for building'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.21.13
+
+    - name: Pull Docker image
+      run: docker pull ${{ inputs.docker_image }}
+      shell: bash
+
+    - name: Build and package using Docker
+      run: |
+        docker run --rm \
+          -e TARGET_OS=${{ inputs.target_os }} \
+          -e TARGET_ARCH=${{ inputs.target_arch }} \
+          -v "${{ github.workspace }}:/workspace" \
+          -w /workspace \
+          ${{ inputs.docker_image }} \
+          sh -c "make module"
+      shell: bash
+
+    - name: Verify module.tar.gz exists recursively
+      run: |
+        if find . -name module.tar.gz | grep -q .; then
+          echo "module.tar.gz exists"
+        else
+          echo "module.tar.gz does not exist"
+          exit 1
```

Comment:
so the goal here is just to report if module.tar.gz exists? will this cause the workflow to fail?

Reply:
I originally had this in for debugging but decided to keep it since it made sanity checking the existence of the `make module` `module.tar.gz` artifact easier. I think we can remove though it's not critical